org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/participant/XtextCopyResourceParticipant.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2017, 2020 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.ui.refactoring2.participant;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.ltk.core.refactoring.Change;
+import org.eclipse.ltk.core.refactoring.RefactoringStatus;
+import org.eclipse.ltk.core.refactoring.participants.CheckConditionsContext;
+import org.eclipse.ltk.core.refactoring.participants.CopyArguments;
+import org.eclipse.ltk.core.refactoring.participants.CopyParticipant;
+import org.eclipse.ltk.core.refactoring.participants.ISharableParticipant;
+import org.eclipse.ltk.core.refactoring.participants.RefactoringArguments;
+import org.eclipse.xtext.ide.refactoring.ResourceRelocationContext;
+
+import com.google.inject.Inject;
+
+/**
+ * @author koehnlein - Initial contribution and API
+ * @since 2.13
+ */
+public class XtextCopyResourceParticipant extends CopyParticipant implements ISharableParticipant {
+
+	@Inject
+	private ResourceRelocationProcessor processor;
+
+	private Change change;
+
+	@Override
+	public RefactoringStatus checkConditions(IProgressMonitor pm, CheckConditionsContext context) throws OperationCanceledException {
+		try {
+			change = processor.createChange(getName(), ResourceRelocationContext.ChangeType.COPY, pm);
+			return processor.getIssues().getRefactoringStatus();
+		} catch (OperationCanceledException e) {
+			throw e;
+		} catch (CoreException e) {
```

Comment:
I'm sure this should better become part of the result, e.g. something like RefactoringStatus .FATAL should be put in use here.

Reply:
I can do `return RefactoringStatus.createFatalErrorStatus("Error copying resource participant");` . @szarnekow @cdietrich Is the ok to have the above error message?

Reply:
You could use RefactoringStatus.create(coreException.getStatus)